tests/test_transformers.py

```diff
@@ -89,3 +89,23 @@ def test_transformers_raise_notfitted_transform(transformer):
     X, y = load_moscow_stjoes(return_X_y=True)
     with pytest.raises(NotFittedError):
         transformer().transform(X)
+
+
+@pytest.mark.parametrize("transformer", [CCATransformer, CCorATransformer])
+@pytest.mark.parametrize("n_components", [None, 0, 5])
+def test_transformers_n_components(transformer, n_components):
+    """Test that n_components is handled correctly."""
+    X, y = load_moscow_stjoes(return_X_y=True)
+    t = transformer(n_components=n_components).fit(X, y)
+    assert t._n_features_out == t.n_components_
+    assert t.transform(X).shape[1] == t.n_components_
+
+
+@pytest.mark.parametrize("transformer", [CCATransformer, CCorATransformer])
+@pytest.mark.parametrize("n_components", [-1, 1000])
```

Comment:
> For the second, I'm just testing a negative value and an absurdly high value which both fall out of range, but not psyched about the constant values again.

I get the uneasiness, but I think the choice of constants is a little clearer here than in the first test, especially since the docstring basically explains that these are out of range. I'm okay with this as-is.